### PR TITLE
quiche: enforce header verification in QUICHE

### PR DESCRIPTION
**Pull request by danzh2010**
Signed-off-by: Dan Zhang <danzh@google.com>

Commit Message: make QUICHE reject invalid request/response headers

Risk Level: low
Testing: added integration test

### CODE REVIEW

File: test/integration/protocol_integration_test.cc [hunk 1]
```diff
@@ -3573,4 +3573,47 @@ TEST_P(DownstreamProtocolIntegrationTest, HandleSocketFail) {
   test_server_.reset();
 }
 
+#ifdef NDEBUG
```
Comment (wrowe) [inline test/integration/protocol_integration_test.cc:3576]: Why not test in all cases?

**Review by wrowe — APPROVED** [review]
lgtm modulo one question

**Review by wrowe — CHANGES_REQUESTED** [review]
You will need to modify the tests;
//test/integration:quic_protocol_integration_test
//test/integration:quic_http_integration_test

to account for the new, proper behavior in quic header handling.

**Comment by danzh2010** [comment]
There is still an upstream QUICHE change needed. I'll ping back once this PR is ready for review. Thanks!

**Pull request closed without merging**